### PR TITLE
DOC: Enforce Numpy Docstring Validation for pandas.HDFStore.info

### PR DESCRIPTION
**Pull request by tuhinsharma121**
- [ ] xref #58067

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -1688,23 +1688,26 @@ def info(self) -> str:
...
+        >>> store.put("data1", df1)  # doctest: +SKIP
+        >>> store.put("data2", df2)  # doctest: +SKIP
         >>> print(store.info())  # doctest: +SKIP
         >>> store.close()  # doctest: +SKIP
         <class 'pandas.io.pytables.HDFStore'>
         File path: store.h5
-        /data    frame    (shape->[2,2])
+        /data1            frame        (shape->[2,2])
+        /data2            frame        (shape->[2,2])
         """
```
Comment (tuhinsharma121) [inline pandas/io/pytables.py:1711]: @mroeschke I took the liberty to refine the example by adding one more dataframe. Let me know if this is good or I need to revert back.

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -1688,17 +1688,26 @@ def info(self) -> str:
         Returns
         -------
         str
+            A String containing the python pandas class name, filepath to the HDF5
+            file and all the object keys along with their respective dataframe shapes.
+
+        See Also
+        --------
+        HDFStore.get_storer : Returns the storer object for a key.
```
Comment (tuhinsharma121) [inline pandas/io/pytables.py:1697]: @mroeschke I could not find a closer method than this.

**Review by mroeschke — APPROVED** [review]

File: pandas/io/pytables.py (hunk 1 above)
Comment (mroeschke) [reply pandas/io/pytables.py:1711]: Sure thanks for the addition

**Comment by mroeschke** [comment]
Thanks @tuhinsharma121

**Pull request merged**